core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -164,13 +164,13 @@ private[server] class MetadataCache(brokerId: Int) extends Logging {
       aliveNodes.clear()
       aliveBrokers.clear()
       updateMetadataRequest.liveBrokers.asScala.foreach { broker =>
-        val nodes = new EnumMap[SecurityProtocol, Node](classOf[SecurityProtocol])
-        val endPoints = new EnumMap[SecurityProtocol, EndPoint](classOf[SecurityProtocol])
-        broker.endPoints.asScala.foreach { case (protocol, ep) =>
-          endPoints.put(protocol, EndPoint(ep.host, ep.port, protocol))
-          nodes.put(protocol, new Node(broker.id, ep.host, ep.port))
+        val nodes = new java.util.HashMap[ListenerName, Node]
```

Comment:
We optimised this code quite a bit in a previous release as it affects the performance of metadata requests. The default Scala HashMap is a bit slower than Java's HashMap. In Scala 2.11, there's `AnyRefMap` that is comparable to Java's, but we still support 2.10.